gc-worker/server/server.go

```diff
@@ -0,0 +1,302 @@
+// Copyright 2022 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/docker/go-units"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	pd "github.com/tikv/pd/client"
+	"github.com/tikv/pd/pkg/tsoutil"
+	"go.etcd.io/etcd/clientv3"
+	"go.etcd.io/etcd/clientv3/concurrency"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+)
+
+const (
+	etcdTimeout = time.Duration(3) * time.Second
+	// etcdElectionPath for all gcworker servers.
+	etcdElectionPath = "/gc-worker/election"
+	etcdElectionVal  = "local"
+	maxPdMsgSize     = int(128 * units.MiB)
+)
+
+// LogPDInfo prints the PD version information.
+func LogGCWorkerInfo() {
+	log.Info("Welcome to GC-Worker for TiKV.")
+}
+
+// PrintPDInfo prints the PD version information without log info.
+func PrintGCWorkerInfo() {
+	fmt.Printf("GC Worker for TiKV version 0.1.0.\n")
```

Comment:
Nit: set version by Makefile. Refer to [here](https://github.com/pingcap/tidb/blob/4b917c47ecbce19e098a0eef19609742bddec7da/Makefile.common#L63).

Reply:
done. version info seems much more complicate. Here just catch a simple git version. May improve this later.